core/arch/arm/mm/tee_pager.c

```diff
@@ -201,84 +211,124 @@ static void pager_unlock(uint32_t exceptions)
 
 void *tee_pager_phys_to_virt(paddr_t pa)
 {
-	struct core_mmu_table_info *ti = &tee_pager_tbl_info;
+	struct core_mmu_table_info ti;
 	unsigned idx;
-	unsigned end_idx;
 	uint32_t a;
 	paddr_t p;
+	vaddr_t v;
+	size_t n;
 
-	end_idx = core_mmu_va2idx(ti, CFG_TEE_RAM_START +
-				      CFG_TEE_RAM_VA_SIZE);
-	/* Most addresses are mapped lineary, try that first if possible. */
-	idx = core_mmu_va2idx(ti, pa);
-	if (idx >= core_mmu_va2idx(ti, CFG_TEE_RAM_START) &&
-	    idx < end_idx) {
-		core_mmu_get_entry(ti, idx, &p, &a);
-		if ((a & TEE_MATTR_VALID_BLOCK) && p == pa)
-			return (void *)core_mmu_idx2va(ti, idx);
-	}
+	/*
+	 * Most addresses are mapped lineary, try that first if possible.
+	 */
+	if (!tee_pager_get_table_info(pa, &ti))
+		return NULL; /* impossible pa */
```

Comment:
Actually the comment applies to the call of `tee_pager_get_table_info()` too.
I'll skip the cast as frivolous casting causes more problems than they solve.